src/Sylius/Component/Core/Model/ChannelPricing.php

```diff
@@ -86,6 +86,11 @@ public function isPriceReduced(): bool
 
     public function addAppliedPromotion(array $promotion): void
     {
+        if ($this->appliedPromotions === null) {
+            $this->appliedPromotions = $promotion;
+            return;
+        }
+
         $this->appliedPromotions = array_merge_recursive($this->appliedPromotions, $promotion);
     }
```

Comment:
We should consider some changes for 
`/** @var array */
    protected $appliedPromotions = [];`
value. 
At this moment it does not safe us from storing null value. This lead to problem with array_merge where one value sometimes is null. 
Sugestion: change this to `protected array $appliedPromotions = [];` make adjust in orm file and create the migration that will make this field not null and put default value of empty array.